libs/dkg/src/dkg.cpp

```diff
@@ -754,7 +719,53 @@ bool DistributedKeyGeneration::BuildQual()
 }
 
 /**
- * If in qual a member Computes individual share of the secret key and further computes and
+ * Checks coefficients sent by qual members and puts their address and the secret shares we received
+ * from them into a complaints maps if the coefficients are not valid
+ *
+ * @return Map of address and pair of secret shares for each qual member we wish to complain against
+ */
+std::unordered_map<
+    DistributedKeyGeneration::MuddleAddress,
+    std::pair<DistributedKeyGeneration::MsgShare, DistributedKeyGeneration::MsgShare>>
+DistributedKeyGeneration::ComputeQualComplaints()
+{
+  std::unordered_map<MuddleAddress, std::pair<MsgShare, MsgShare>> qual_complaints;
+
+  uint32_t i  = 0;
+  auto     iq = cabinet_.begin();
+  for (auto const &miner : qual_)
+  {
+    while (*iq != miner)
+    {
+      ++iq;
+      ++i;
+    }
```

Comment:
Minor: This code seems like you could use std::distance

Reply:
Yes, but using std::distance would mean having to start from the beginning of cabinet_ every time to find the index i for the miner in qual.

Reply:
Yes, but then you would have to go through the same distance in cabinet_ twice: once to find the miner inside cabinet_ (which is what iq does) and then a second time to compute the index of the miner in cabinet. In the while loop both iq and i are iterated at the same time so we only have to traverse cabinet_ once. There might be a nicer way to do it though.